libs/partners/milvus/langchain_milvus/vectorstores/milvus.py

```diff
@@ -738,9 +738,69 @@ def add_texts(
         Returns:
             List[str]: The resulting keys for each inserted element.
         """
-        from pymilvus import Collection, MilvusException
 
         texts = list(texts)
+
+        try:
+            embeddings: list = self.embedding_func.embed_documents(texts)
+        except NotImplementedError:
+            embeddings = [self.embedding_func.embed_query(x) for x in texts]
+
+        if len(embeddings) == 0:
+            logger.debug("Nothing to insert, skipping.")
+            return []
+
+        return self.add_embeddings(
+            text_embeddings=zip(texts, embeddings),
+            metadatas=metadatas,
+            timeout=timeout,
+            batch_size=batch_size,
+            ids=ids,
+            **kwargs,
+        )
+
+    def add_embeddings(
+        self,
+        text_embeddings: Iterable[Tuple[str, List[float]]],
```

Comment:
Should probably be

```suggestion
        embeddings: Sequence[Tuple[str, List[float]]],
```

Reply:
This is just a drive by review. I also don't know if it makes more sense to add embeddings as an optional argument to add_texts to avoid proliferation of entry points (we should see what's usually done for other vectorstores).

Reply:
I think it would be clearer to separate the functionality and add a separate method instead of adding just another parameter.

I did some research and some of the vector stores implementation in langchain has already `add_embeddings` method:
 - [Elasticsearch](https://github.com/langchain-ai/langchain-elastic/blob/main/libs/elasticsearch/langchain_elasticsearch/vectorstores.py#L1091-#L1128)
 - [Postgres](https://github.com/langchain-ai/langchain-postgres/blob/main/langchain_postgres/vectorstores.py#L747-#L803)
 - [langchain-community] [ScaNN](https://github.com/langchain-ai/langchain/blob/de70a64e3a914a71aa875195b8171f559e13e5ca/libs/community/langchain_community/vectorstores/scann.py#L108-#L134)
 - [langchain-community] [FAISS](https://github.com/langchain-ai/langchain/blob/de70a64e3a914a71aa875195b8171f559e13e5ca/libs/community/langchain_community/vectorstores/faiss.py#L363-#L383)
 - [langchain-community] [`Azure Cognitive Search` vector store.](https://github.com/langchain-ai/langchain/blob/de70a64e3a914a71aa875195b8171f559e13e5ca/libs/community/langchain_community/vectorstores/azuresearch.py#L484-#L543)
 - [langchain-community] [`Kinetica` store](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/kinetica.py#L318-#L348) 
 - [langchain-community] [`Postgres` with the `lantern` extension as a vector store.](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/lantern.py#L467-#L484)
 - [langchain-community] [`Neo4j` vector index](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/neo4j_vector.py#L861-#L906)
 - [langchain-community] [`Amazon OpenSearch Vector Engine` vector store.](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/opensearch_vector_search.py#L645-#L680)
 - [langchain-community] [`Postgres` with the `pg_embedding` extension as a vector store.](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/pgembedding.py#L260-#L281)

Only a Postgres, Kinetica and Neo4j has method signature:
```python
def add_embeddings(
        self,
        texts: Iterable[str],
        embeddings: List[List[float]],
        metadatas: Optional[List[dict]] = None,
        ids: Optional[List[str]] = None,
        **kwargs: Any,
    ) -> List[str]: ...
```

and the rest, (including Elastic!) has the signature

```python
def add_embeddings(
        self,
        text_embeddings: Iterable[Tuple[str, List[float]]],
        metadatas: Optional[List[dict]] = None,
        **kwargs: Any
    ) -> List[str]: ....

```